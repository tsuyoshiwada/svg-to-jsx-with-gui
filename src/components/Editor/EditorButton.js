import React, { Component } from 'react';
import styled from 'styled-components';
import { palette } from '../../styles';
import Button from '../internal/Button';


const EditorButtonWrapper = Button.extend`
  padding: 0 20px;
  height: 40px;
  border-left: 1px solid rgba(255, 255, 255, 0.1);
  background: ${palette.secondaryDark};
  color: #fff;
  font-size: 0.75rem;

  &:hover {
    background-color: ${palette.secondaryLight};
  }

  &:active {
    background-color: #393f5d;
  }
`;

const IconWrapper = styled.span`
  display: inline-block;
  width: 1em;
  height: 1em;
  margin: 0 0.5em 0 0;

  & svg {
    width: 1em;
    height: 1em;
    fill: #fff;
  }
`;


export default class EditorButton extends Component {
  handleClick = (e) => {
    e.preventDefault();
    if (typeof this.props.onClick === 'function') {
      this.props.onClick(e);
    }
  };

  render() {
    const {
      children,
      icon,
      onClick,
      ...rest
    } = this.props;

    return (
      <EditorButtonWrapper
        {...rest}
        onClick={this.handleClick}
      >
        {icon &&
          <IconWrapper>{icon}</IconWrapper>
        }
        {children}
      </EditorButtonWrapper>
    );
  }
}
