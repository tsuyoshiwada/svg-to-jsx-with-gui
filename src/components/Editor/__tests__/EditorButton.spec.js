import React from 'react';
import { shallow } from 'enzyme';
import EditorButton from '../EditorButton';


describe('components#<EditorButton />', () => {
  test('Should be render', () => {
    const wrapper = shallow(
      <EditorButton>
        Foo
      </EditorButton>,
    );

    expect(wrapper).toMatchSnapshot();
  });


  test('Should be render icon', () => {
    const wrapper = shallow(
      <EditorButton
        icon={
          <svg width={10} height={10}>
            <title>icon</title>
          </svg>
        }
      >
        Bar
      </EditorButton>,
    );

    expect(wrapper).toMatchSnapshot();
  });


  test('Should be call onClick when fired click event', () => {
    const fn = jest.fn();
    const wrapper = shallow(<EditorButton onClick={fn} />);

    expect(fn.mock.calls.length).toBe(0);

    wrapper.simulate('click', { preventDefault: () => null });

    expect(fn.mock.calls.length).toBe(1);
  });
});
