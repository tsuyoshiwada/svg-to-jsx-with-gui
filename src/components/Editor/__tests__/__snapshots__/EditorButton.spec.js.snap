// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components#<EditorButton /> Should be render 1`] = `
<styled.button
  onClick={[Function]}
>
  Foo
</styled.button>
`;

exports[`components#<EditorButton /> Should be render icon 1`] = `
<styled.button
  onClick={[Function]}
>
  <styled.span>
    <svg
      height={10}
      width={10}
    >
      <title>
        icon
      </title>
    </svg>
  </styled.span>
  Bar
</styled.button>
`;
